src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ProjectsControllerTests.kt

```diff
@@ -145,29 +142,54 @@ class ProjectsControllerTests {
     }
 
     @Test
-    fun `can delete version`()
-    {
+    fun `can clone project to user`() {
+        val mockVersionRepo = mock<VersionRepository>()
+        val mockProjectRepo = mock<ProjectRepository> {
+            on { getProject(1) } doReturn Project(1, "p1", listOf(Version("v1", "createdTime", "updatedTime", 1),
+                    Version("v2", "createdTime", "updatedTime", 1)))
+            on { saveNewProject("uid", "p1") } doReturn 2
+        }
+        val mockLogic = mock<UserLogic> {
+            on { getUser("new.user@email.com") } doReturn CommonProfile().apply { id = "uid" }
+        }
+        val sut = ProjectsController(mockSession, mockVersionRepo, mockProjectRepo, mockLogic)
+        sut.cloneProjectToUser(1, "new.user@email.com")
+        verify(mockVersionRepo).cloneVersion("v1", "testVersion", 2)
+        verify(mockVersionRepo).cloneVersion("v2", "testVersion", 2)
+    }
+
+    @Test
+    fun `clone project to user throws if user does not exist`() {
```

Comment:
I'm pretty confused about codecov complaining because I believe this test hits the condition it's looking for?

Reply:
Looks that way to me too!